### PR TITLE
Use max rho of surface to compute A(z)

### LINKED ISSUE #1083: A(z) computed wrong on grids without lcfs
Take the surface in `test_compute_everything` and add rho=0.8 to the constructor. `FourierRZToroidalSurface(**elliptic_cross_section_with_torsion, rho=0.8)`. Then the grid only has rho =0.8 nodes and A(z) is computed as zero. I think the fix is to compute this like we do `perimeter(z)`; fix_surface to the rho value of the surface and then divide by that rho value to get an approximation, but I'll take a closer look tomorrow.

_Originally posted by @unalmis in https://github.com/PlasmaControl/DESC/pull/1024#discussion_r1656635137_

### PR DESCRIPTION
**Pull request by unalmis**
Resolves #1083

### CODE REVIEW

File: desc/compute/_geometry.py [hunk 1]
```diff
@@ -196,7 +196,7 @@ def _A_of_z_FourierRZToroidalSurface(params, transforms, profiles, data, **kwarg
             transforms["grid"],
             data["Z"] * n[:, 2] * jnp.sqrt(data["g_tt"]),
             line_label="theta",
-            fix_surface=("rho", 1.0),
```
Comment (f0uriest) [inline desc/compute/_geometry.py:199]: do we also want to scale the area up if the largest rho isnt 1?

Comment (unalmis) [reply desc/compute/_geometry.py:199]: I don't think that makes sense. It's proportional sure but the integrand will be different, so it's not exact.

File: tests/inputs/master_compute_data.pkl
Comment (unalmis) [inline tests/inputs/master_compute_data.pkl:1]: Changed from adding e_theta|r,p. A(z) is same for rho=1 surface.

File: desc/compute/_geometry.py (hunk 1 above)
Comment (unalmis) [reply desc/compute/_geometry.py:199]: Was our justification to do that for `perimeter(z)` because we assume integrand doesn't change much near edge of quadrature node and the lcfs?

File: desc/compute/_geometry.py [hunk 2]
```diff
@@ -176,27 +176,38 @@ def _A_of_z(params, transforms, profiles, data, **kwargs):
...
+    data=["Z", "n_rho", "e_theta|r,p"],
     parameterization=["desc.geometry.surface.FourierRZToroidalSurface"],
+    # FIXME: Add source grid requirement once omega is nonzero.
 )
 def _A_of_z_FourierRZToroidalSurface(params, transforms, profiles, data, **kwargs):
-    # divergence theorem: integral(dA div [0, 0, Z]) = integral(ds n dot [0, 0, Z])
-    # but we need only the part of n in the R,Z plane
+    # Denote any vector v = [vᴿ, v^ϕ, vᶻ] with a tuple of its contravariant components.
+    # We use a 2D divergence theorem over constant ϕ toroidal surface (i.e. R, Z plane).
+    # Denote the divergence operator div by [∂_R, ∂_ϕ, ∂_Z] ⊗ [1, 0, 1].
```
Comment (unalmis) [inline desc/compute/_geometry.py:193]: in the RZ plane divergence operator is cartesian

File: desc/compute/_geometry.py [hunk 3]
```diff
@@ -176,27 +176,39 @@ def _A_of_z(params, transforms, profiles, data, **kwargs):
...
-    description="Cross-sectional area as function of zeta",
+    description="Enclosed cross-sectional (constant phi surface) area, "
+    "as function of zeta",
     dim=1,
     params=[],
     transforms={"grid": []},
     profiles=[],
     coordinates="z",
-    data=["Z", "n_rho", "g_tt"],
+    data=["Z", "n_rho", "e_theta|r,p", "rho"],
```
Comment (dpanici) [inline desc/compute/_geometry.py:186]: why this change? is this for when omega is nonzero?

File: desc/compute/_basis_vectors.py [hunk 4]
```diff
@@ -3523,3 +3523,25 @@ def _n_zeta(params, transforms, profiles, data, **kwargs):
...
+    profiles=[],
+    coordinates="rtz",
+    data=["e_theta", "e_phi", "phi_t"],
+    parameterization=[
+        "desc.equilibrium.equilibrium.Equilibrium",
+        "desc.geometry.surface.FourierRZToroidalSurface",
+    ],
+)
+def _e_sub_theta_rp(params, transforms, profiles, data, **kwargs):
+    data["e_theta|r,p"] = data["e_theta"] - (data["e_phi"].T * data["phi_t"]).T
```
Comment (dpanici) [inline desc/compute/_basis_vectors.py:3546]: `e_theta` by definition a partial derivative at constant rho and phi right? I do not quite get this expression

Comment (unalmis) [reply desc/compute/_basis_vectors.py:3546]: No, it's constant rho and zeta

File: desc/compute/_geometry.py (hunk 3 above)
Comment (unalmis) [reply desc/compute/_geometry.py:186]: Yes, I was checking the math for this to be sure, and i was gonna need to do this anyway so I did it here

**Review by dpanici — APPROVED** [review]

File: desc/compute/_basis_vectors.py [hunk 5]
```diff
@@ -3523,3 +3523,25 @@ def _n_zeta(params, transforms, profiles, data, **kwargs):
...
     )
     return data
+
+
+@register_compute_fun(
+    name="e_theta|r,p",
+    label="\\mathbf{e}_{\\theta} |_{\\rho, \\phi}",
+    units="m",
+    units_long="meters",
+    description="Tangent vector along boundary of constant phi toroidal surface",
```
Comment (f0uriest) [inline desc/compute/_basis_vectors.py:3533]: Is this the most accurate description?

File: desc/compute/_geometry.py (hunk 1 above)
Comment (f0uriest) [reply desc/compute/_geometry.py:199]: Yeah pretty much. Obviously if you're using the rho=0.5 surface it's a bad guess, but for rho near 1 it should be reasonably accurate (and exact for elliptical cross section)

File: desc/compute/_basis_vectors.py (hunk 5 above)
Comment (unalmis) [reply desc/compute/_basis_vectors.py:3533]: It is precise. i'll add the standard "covariant vector ..."

**Review by f0uriest — APPROVED** [review]

**Review by dpanici — APPROVED** [review]
